Source/Core/Core/PowerPC/PPCSymbolDB.cpp

```diff
@@ -308,132 +415,342 @@ bool PPCSymbolDB::LoadMap(const std::string& filename, bool bad)
     const char* namepos = strstr(line, name);
     if (namepos != nullptr)  // would be odd if not :P
       strcpy(name, namepos);
     name[strlen(name) - 1] = 0;
     if (name[strlen(name) - 1] == '\r')
       name[strlen(name) - 1] = 0;
 
-    // Check if this is a valid entry.
-    if (strlen(name) > 0)
+    // Later CodeWarrior Symbol Maps record the padding between compilation units with symbols named
+    // "*fill*".  If something special needs to be done with these symbols in the future, this would
+    // be the place to do so.
+    if (!strcmp(name, "*fill*"))
     {
-      // Can't compute the checksum if not in RAM
-      bool good = !bad && PowerPC::HostIsInstructionRAMAddress(vaddress) &&
-                  PowerPC::HostIsInstructionRAMAddress(vaddress + size - 4);
-      if (!good)
+      fill_symbol_count++;
+    }
+
+    // Q&A: What is a Closure Symbol?
+    // Closure Symbols are symbols created by the compiler(?) that envelope all symbols in a
+    // section from the same compilation unit. They come in two variants, which I've dubbed
+    // "Start" and "End" Closure Symbols. Static Libraries (*.a), when linked, only generate Start
+    // Closure Symbols.  Objects (*.o), when linked, generate both Start and End Closure Symbols
+    // for text sections, and only Start Closure Symbols for everything else. Start Closure
+    // Symbols typically have names such as ".text" or ".sdata". End Closure Symbols typically
+    // have names such as
+    // ".text.44391". Closure Symbols almost always have a reported alignment of 1 byte.
+
+    // Closure Symbols are not useful with how Dolphin's Symbol Database works.  Start Closure
+    // Symbols overlap the entire compilation unit, which interferes with disassembly
+    // highlighting. Or, it would, if it weren't for the fact that the function "AddKnownSymbol"
+    // called later in this function is guaranteed to overwrite any Closure Symbols, as they have
+    // the same virtual address as the first symbol of the compilation unit they represent.
+    // Furthermore, the reported size of Start Closure Symbols includes those of Unused Symbols
+    // stripped from the binary, making them inaccurate for usage at runtime.  Thus, the following
+    // check formalizes the removal of Closure Symbols until they are more useful / less harmful
+    // to Dolphin.
+
+    // The ctors and dtors sections don't have traditional Closure Symbols. Also, some symbol maps
+    // have extra linker generated closure symbols in these sections.  These symbols have oddities
+    // such as "$00" to "$99" tacked on the end of the name, '_' instead of '.' as the first
+    // character, a reported alignment of 4 bytes, and maybe more?  Basically, ctors and dtors are
+    // left as is, because they are too weird.
+
+    if (alignment == 1)
+    {
+      if ((!strncmp(name, ".init", 5)) || (!strncmp(name, "extab", 5)) ||
```

Comment:
These should probably be based on the section name and not hardcoded.

Reply:
Done.  This will adversely affect any symbol maps that were loaded into Dolphin and then resaved (section names won't match anymore), but honestly those symbol maps aren't worth worrying about anyways.